docs/source/api/plugin/inline-trace.mdx

```diff
@@ -0,0 +1,142 @@
+---
+title: "API Reference: Inline trace plugin"
+api_reference: true
+---
+
+## Using the plugin
+
+> 📣 **New in Apollo Server 4**: error details are not included in traces by default. For more details, see [Error Handling](../../data/errors/#masking-and-logging-errors).
+
+This article documents the options for the `ApolloServerPluginInlineTrace` plugin, which you can import from `@apollo/server/plugin/inlineTrace`.
+
+This plugin enables your GraphQL server to include encoded performance and usage traces inside responses. This is primarily designed for use with [Apollo Federation](/federation/metrics/). Federated subgraphs use this plugin and include a trace in the `ftv1` GraphQL response extension if requested to do so by the Apollo gateway. The gateway requests this trace by passing the HTTP header `apollo-federation-include-trace: ftv1`.
+
+Apollo Server installs this plugin by default in all federated subgraphs, with its default configuration. Apollo Server check whether it it is a federated subgraph if the schema it is serving includes a field `_Service.sdl: String`.  You typically do not have to install this plugin yourself; you only need to do so if you want to provide non-default configuration.
+
+If you want to configure the `ApolloServerPluginInlineTrace` plugin, import it and pass it to your `ApolloServer` constructor's `plugins` array:
+
+<MultiCodeBlock>
+
+```ts
+import { ApolloServer } from '@apollo/server';
+import { ApolloServerPluginInlineTrace } from '@apollo/server/plugin/inlineTrace';
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  plugins: [
+    ApolloServerPluginInlineTrace({
+      includeErrors: { transform: (err) => (err.message.match(SENSITIVE_REGEX) ? null : err ) },
+    }),
+  ],
+});
+```
+
+</MultiCodeBlock>
+
+If you don't want to use the inline trace plugin even though your schema defines `_Service.sdl: String`, you can explicitly disable it with the `ApolloServerPluginInlineTraceDisabled` plugin:
+
+```js
```

Comment:
ts / MultiCodeBlock here?

Reply:
Oh oops! That one slipped by me, thanks!